orbit/pkg/update/execwinapi_windows.go

```diff
@@ -0,0 +1,122 @@
+//go:build windows
+
+package update
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
+)
+
+var (
+	dllMDMRegistration *windows.LazyDLL = windows.NewLazySystemDLL("mdmregistration.dll")
+
+	// RegisterDeviceWithManagement registers a device with a MDM service:
+	// https://learn.microsoft.com/en-us/windows/win32/api/mdmregistration/nf-mdmregistration-registerdevicewithmanagement
+	procRegisterDeviceWithManagement *windows.LazyProc = dllMDMRegistration.NewProc("RegisterDeviceWithManagement")
+)
+
+// Exported so that it can be used in tools/ (so that it can be built for
+// Windows and tested on a Windows machine). Otherwise not meant to be called
+// from outside this package.
+func RunMicrosoftMDMEnrollment(args MicrosoftMDMEnrollmentArgs) error {
+	installType, err := readInstallationType()
+	if err != nil {
+		return err
+	}
+	if strings.ToLower(installType) == "server" {
+		// do not enroll, it is a server
+		return errIsWindowsServer
+	}
+	return enrollHostToMDM(args)
+}
+
+func readInstallationType() (string, error) {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
+	if err != nil {
+		return "", err
+	}
+	defer k.Close()
+
+	s, _, err := k.GetStringValue("InstallationType")
+	if err != nil {
+		return "", err
+	}
+	return s, nil
+}
+
+// TODO(mna): refactor to a Windows-specific package to constrain usage of
+// unsafe to that package, once https://github.com/fleetdm/fleet/pull/12387
+// lands.
```

Comment:
Left here for now, as it would touch a number of different places in the code. Best to do it as part of a PR just for this change.